### PR TITLE
Decrease animation duration

### PR DESCRIPTION
**Pull request by george-lim**
NOTE: Instead of storing the CGPaths in a variable in TBSwitch.m, when the CGPaths are directly assigned to the layer, it seems to decrease significant lag on the switch. Perhaps the CGPaths aren't being freed properly?

### CODE REVIEW

File: TBSwitch/TBSwitch.m [hunk 1]
```diff
@@ -64,7 +64,8 @@ - (id)initWithCoder:(NSCoder *)aDecoder {
   if( (self = [super initWithCoder:aDecoder]) ){
     self.backgroundImageView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"switch_bg_small"]];
     self.backgroundImageView.contentMode = UIViewContentModeScaleAspectFit;
-    self.backgroundImageView.image = [self.backgroundImageView.image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
+    self.backgroundImageView.image =
+    [self.backgroundImageView.image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
```
Comment (svintus) [inline TBSwitch/TBSwitch.m]: Please fix indentation here.

File: TBSwitch/TBSwitch.m [hunk 2]
```diff
@@ -64,7 +64,8 @@ - (id)initWithCoder:(NSCoder *)aDecoder {
...
     if (self.on)
```
Comment (svintus) [inline TBSwitch/TBSwitch.m]: Let's take this opportunity to update the code to conform to our Obj-C style guide. If statements should always have braces.

File: TBSwitch/TBSwitch.m [hunk 3]
```diff
@@ -146,13 +147,8 @@ - (void)configureThumbLayer
...
   self.thumbLayer.shadowOpacity = 0.3f;
-  
-  // TODO: use proper circular shape
-  CGPathRef knobPath =
-  [UIBezierPath
-   bezierPathWithRoundedRect: self.thumbLayer.bounds
-   cornerRadius: floorf(self.thumbLayer.bounds.size.height/2.0)].CGPath;
-  self.thumbLayer.path = knobPath;
+  self.thumbLayer.path =
+  [UIBezierPath bezierPathWithOvalInRect:self.thumbLayer.bounds].CGPath;
```
Comment (svintus) [inline TBSwitch/TBSwitch.m]: Same here - when an assignment is split into multiple lines, lines after the 1st should have one indent.

File: TBSwitch/TBSwitch.m [hunk 4]
```diff
@@ -186,20 +182,15 @@ - (void)configureKnob
...
       self.knobLayer.frame = self.thumbLayer.bounds;
       self.knobLayer.cornerRadius = self.thumbLayer.frame.size.height/2.0;
-      // TODO: use proper circular shape
-      CGPathRef knobPath =
-      [UIBezierPath
-       bezierPathWithRoundedRect: self.knobLayer.bounds
-       cornerRadius: floorf(self.knobLayer.bounds.size.height/2.0)].CGPath;
-      self.knobLayer.path = knobPath;
+      self.knobLayer.path =
+      [UIBezierPath bezierPathWithOvalInRect:self.knobLayer.bounds].CGPath;
```
Comment (svintus) [inline TBSwitch/TBSwitch.m]: and here

File: TBSwitch/TBSwitch.m [hunk 5]
```diff
@@ -186,20 +182,15 @@ - (void)configureKnob
...
-      CAMediaTimingFunction *secondLinearCurve =
-      [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionEaseOut];
       CABasicAnimation *dismissAnimation =
       [CABasicAnimation animationWithKeyPath:@"transform.scale"];
       dismissAnimation.toValue = @0;
-      dismissAnimation.duration = 0.55;
-      dismissAnimation.timingFunction = secondLinearCurve;
+      dismissAnimation.duration = 0.2;
+      dismissAnimation.timingFunction =
+      [CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionEaseOut];
```
Comment (svintus) [inline TBSwitch/TBSwitch.m]: and here :)

File: TBSwitch/TBSwitch.m [hunk 6]
```diff
@@ -13,7 +13,7 @@
 #define IS_IPAD() (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad)
 #define THUMB_MARGIN 3
 #define ORANGE_COLOR [UIColor colorWithRed:237/255.0f green:175/255.0f blue:25/255.0f alpha:1]
-#define SWITCH_SPEED 0.4
+#define SWITCH_SPEED 0.1
```
Comment (codyleblanc) [inline TBSwitch/TBSwitch.m]: did you get designs opinion on this?

Comment (codyleblanc) [reply TBSwitch/TBSwitch.m]: Or any of the speed changes?

Comment (svintus) [reply TBSwitch/TBSwitch.m]: The speed should match macOS now. @george-lim can you confirm?

Comment (george-lim) [reply TBSwitch/TBSwitch.m]: yup

**Pull request merged**